config/src/config/network_config.rs

```diff
@@ -157,10 +149,6 @@ impl NetworkConfig {
             inbound_rate_limit_config: None,
             outbound_rate_limit_config: None,
             max_message_size: MAX_MESSAGE_SIZE,
-            inbound_rx_buffer_size_bytes: Some(INBOUND_TCP_RX_BUFFER_SIZE),
```

Comment:
Silly question, but, instead of removing support for this entirely, is there a way to just set the default to not override? e.g., I'm just wondering if there's ever a scenario where we want to set these things?

Reply:
re-adding tcp socket option setting would have the annoying work of figuring out where it goes, so leaving this on but disabled (it already had Option if-None-do-nothing) makes sense to me

Reply:
I honestly don't think we want to mess with this and so might as well clean up the configs now. For this to be effective the user would have to bump up the linux configs, otherwise it has a detrimental effect. 

If you feel strongly I'll leave the config but add a comment on the ways it shouldn't be used.

Reply:
Leave it in or at least leave a comment `// TODO: if needed, set connection socket options here`; _someone_ is going to want to tinker with this again, even if Linux has pretty good defaults

Reply:
Thanks @JoshLind @brianolson . Makes sense, I guess there are other socket options that someone might want to play with in the future as well. I've changed the PR to just set the defaults as `None` and put up a big comment to proceed with caution.